src/base/StrBuf.cpp

```diff
@@ -0,0 +1,49 @@
+#include "StrBuf.hpp"
+#include <memory.h>
+
+//--------------------------------------------------------------------------------
+StrBuf::StrBuf(size_t capacity, char* buffer)
+  : _buf(buffer)
```

Comment:
please avoid identifiers which start with underscore, even though they are locals.
Our coding convention does not allow it.

Reply:
until a final decision regarding naming for private data members for classes, I'll use: m_data_member